src/controllers/movie.js

```diff
@@ -0,0 +1,108 @@
+const {
+    getMoviesDB,
+    createMovieDB,
+    getMovieByIdDB,
+    getMovieByTitleDB,
+    updateMovieDB,
+} = require("../domains/movie.js");
+const { get } = require("../routers/movie.js");
+
+const getMovies = async (req, res) => {
+    const filter = {
+        title: req.query.title,
+        runtimeMins: req.query.runtimeMins,
+        runtimeLt: req.query.runtimeLt,
+        runtimeGt: req.query.runtimeGt,
+    };
+    console.log(filter);
+    const movies = await getMoviesDB(filter);
+    res.status(200).json({ movies: movies });
+};
+
+const createMovie = async (req, res) => {
+    try {
+        const { title, runtimeMins } = req.body;
+        const filter = { title: title };
+        const foundMovie = await getMoviesDB(filter);
+
+        if (foundMovie.length !== 0) {
```

Comment:
`if (!foundMovie.length)`

Reply:
Yup. Same issue where I love to make things as clear as possible for myself. Need to remember to refactor more.

Reply:
I changed this and the tests started failing as it was triggering this guard clause. So I changed it back.